types/normalize-port/index.d.ts

```diff
@@ -0,0 +1,11 @@
+// Type definitions for normalize-port 1.16
+// Project: git@github.com:aoberoi/normalize-port.js.git
+// Definitions by: Alex Alonso <https://github.com/finaldes>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+declare module "normalize-port" {
```

Comment:
This could just be:
```ts
declare function normalizePort(port: any): number;
export = normalizePort;
```

Based on the source code, it looks like this function has interesting behavior:
* If you pass in a number, it will `parseInt` (which does nothing), then return that number if it is `>= 0`, else return `false`. So it is `number => number | false`.
* If you pass in a string, it will `parseInt`, which returns `NaN` on failure. If `parseInt` fails, it will then return the original string. If `parseInt` succeds and returns a positive number, it will return that number. If `parseInt` returns a negative number, it will return `false`. So it is `string => string | number | false`.
* For any other type, `parseInt` might succeed depending on that type's `toString`. So it is `T => T | number | false`.

Depending on which of these behaviors you think are intentional, you could write overloads. I suspect only the first 2 are intentional, so it would be:
```ts
declare function normalizePort(port: number): number | false;
declare function normalizePort(port: string | number): string | number | false;
```
I allowed `string | number` in the second overload, since `normalizePort(number)` returns a subtype of `normalizePort(string)`.